src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Range.cs

```diff
@@ -93,21 +74,27 @@ BoundExpression tryOptimizeOperand(BoundExpression operand)
             }
         }
 
-        private BoundExpression LiftRangeExpression(BoundRangeExpression node, ImmutableArray<BoundExpression> operands)
+        private BoundExpression LiftRangeExpression(BoundRangeExpression node, BoundExpression left, BoundExpression right)
         {
             Debug.Assert(node.Type.IsNullableType());
-            Debug.Assert(operands.Any(operand => operand.Type.IsNullableType()));
-            Debug.Assert(operands.Length == 1 || operands.Length == 2);
+            Debug.Assert(left?.Type.IsNullableType() == true || right?.Type.IsNullableType() == true);
+            Debug.Assert(!(left is null && right is null));
 
-            ArrayBuilder<BoundExpression> sideeffects = ArrayBuilder<BoundExpression>.GetInstance();
-            ArrayBuilder<LocalSymbol> locals = ArrayBuilder<LocalSymbol>.GetInstance();
-            ArrayBuilder<BoundExpression> arguments = ArrayBuilder<BoundExpression>.GetInstance();
+            var sideeffects = ArrayBuilder<BoundExpression>.GetInstance();
+            var locals = ArrayBuilder<LocalSymbol>.GetInstance();
 
-            // left.HasValue && right.HasValue
+            // if (left.HasValue && right.HasValue)
+            //    makeRange(left.GetValueOrDefault(), right.GetValueOrDefault())
             BoundExpression condition = null;
-            foreach (var operand in operands)
+            left = getIndexFromPossibleNullable(left);
+            right = getIndexFromPossibleNullable(right);
+            var rangeExpr = MakeRangeExpression(node.MethodOpt, left, right);
+            BoundExpression getIndexFromPossibleNullable(BoundExpression arg)
```

Comment:
local function in the middle of the method and without spacing is confusing for main flow of logic. Can it be moved at the end of the method?